src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/NodeObservableCollection.cs

```diff
@@ -29,6 +30,16 @@ public bool IsPopulating
             get { return numInits.HasValue && numInits != 0; }
         }
 
+        public bool IsSorted
+        {
+            get
+            {
+                // SMO objects are already sorted so no need to sort them again
+                TreeNode firstITem = this.FirstOrDefault();
+                return firstITem != null && firstITem is SmoTreeNode;

```

Comment:
Minor: I think you can just do `return this.FirstOrDefault() is SmoTreeNode` since it'll return false if casting `null` to `SmoTreeNode`